### PR TITLE
ORKFormItem optionalConvenienceInit

### PR DESCRIPTION
**Pull request by md0u80c9**
Updated the optional form test to confirm all is working.

### CODE REVIEW

File: ResearchKit/Common/ORKFormStep.m [hunk 1]
```diff
@@ -139,6 +139,18 @@ - (void)setFormItems:(NSArray<ORKFormItem *> *)formItems {
...
+        ORKThrowInvalidArgumentExceptionIfNil(identifier);
+        _optional = optional;
+        _identifier = [identifier copy];
+        _text = [text copy];
+        _answerFormat = [answerFormat copy];
+    }
+    return self;
+}
+
 - (instancetype)initWithIdentifier:(NSString *)identifier text:(NSString *)text answerFormat:(ORKAnswerFormat *)answerFormat {
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKFormStep.m]: We might want to reuse your new initializer for this initializer with the default value of `YES` for optional parameter.

File: ResearchKit/Common/ORKFormStep.h [hunk 2]
```diff
@@ -105,6 +105,21 @@ ORK_CLASS_AVAILABLE
...
+ @param optional      Is this item optional?
+ @param answerFormat  The answer format for the form item.
+ 
+ @return An initialized form item.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
+                              text:(nullable NSString *)text
+                          optional:(BOOL) optional
+                      answerFormat:(nullable ORKAnswerFormat *)answerFormat;
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKFormStep.h]: Lets move optional to the end of the init to stay consistent with the init we already have, which follows the pattern `identifier, text, answer format`.

File: Testing/ORKTest/ORKTest/MainViewController.m [hunk 3]
```diff
@@ -1814,8 +1814,8 @@ - (IBAction)miniFormButtonTapped:(id)sender {
         {
             ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:@"fqid_text3"
                                                                    text:@"Text C"
+                                                               optional:NO
                                                            answerFormat:format];
-            item.optional = NO;
```
Comment (umerkhan-apple) [inline Testing/ORKTest/ORKTest/MainViewController.m]: Did you want to update all the other form items that use optional property also? Or did you just do a few to show that it is available?

**Comment by md0u80c9** [comment]
umerkhan-apple. I've made the changes per above. I haven't updated all of ORKTest to use the optional property initialiser just so that we are testing both code paths (I intentionally only changed the 'optional form' test). If yourself and yuanzhu prefer for them all to be changed I can do that.

File: ResearchKit/Common/ORKFormStep.h [hunk 4]
```diff
@@ -105,6 +105,21 @@ ORK_CLASS_AVAILABLE
...
+ Returns an initialized form item using the specified identifier, title, optionality and answer format.
+ 
+ @param identifier    The string that identifies the form item, which should be unique within the form step.
+ @param text          The text displayed as a prompt for the form item's question.
+ @param answerFormat  The answer format for the form item.
+ @param optional      Is this item optional?
+ 
+ @return An initialized form item.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKFormStep.h]: Can you place this initializer after the original one? Just feels like the beginners should see the simple one first.

**Comment by umerkhan-apple** [comment]
@md0u80c9 Looks good Andrew!

@YuanZhu-apple Is this one good to go?

**Comment by YuanZhu-apple** [comment]
Yes, good to go.

**Pull request merged**